chatkit-core/src/test/kotlin/com/pusher/chatkit/users/UserSubscriptionDispatcherTest.kt

```diff
@@ -0,0 +1,193 @@
+package com.pusher.chatkit.users
+
+import com.pusher.chatkit.rooms.api.JoinedRoomApiType
+import com.pusher.chatkit.rooms.api.JoinedRoomApiTypeMapper
+import com.pusher.chatkit.rooms.api.RoomMembershipApiType
+import com.pusher.chatkit.rooms.api.RoomReadStateApiType
+import com.pusher.chatkit.rooms.state.JoinedRoomsState
+import com.pusher.chatkit.rooms.state.roomTwo
+import com.pusher.chatkit.rooms.state.roomTwoId
+import com.pusher.chatkit.state.ChatkitState
+import com.pusher.chatkit.state.JoinedRoom
+import com.pusher.chatkit.state.JoinedRoomsReceived
+import com.pusher.chatkit.state.LeftRoom
+import com.pusher.chatkit.state.RoomDeleted
+import com.pusher.chatkit.state.RoomUpdated
+import com.pusher.chatkit.users.api.UserApiType
+import com.pusher.chatkit.users.api.UserSubscriptionDispatcher
+import com.pusher.chatkit.users.api.UserSubscriptionEvent
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.reduxkotlin.Dispatcher
+import org.reduxkotlin.GetState
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class UserSubscriptionDispatcherTest : Spek({
+
+    val simpleJoinedRoomApiType = JoinedRoomApiType(
+            id = "id1",
+            name = "room1",
+            createdById = "person1",
+            pushNotificationTitleOverride = "notification override",
+            private = false,
+            customData = mapOf("highlight" to "blue"),
+            lastMessageAt = "2020-02-27T17:12:10Z",
+            updatedAt = "2020-02-27T17:12:20Z",
+            createdAt = "2020-02-27T17:12:30Z",
+            deletedAt = null
+    )
+
+    val simpleUser = UserApiType(
+            id = "user1",
+            createdAt = "",
+            updatedAt = "",
+            name = "name",
+            avatarURL = null,
+            customData = null,
+            online = false)
+
+    describe("given a user subscription dispatcher with no previous joined room state") {
+
+        val state = mockk<GetState<ChatkitState>>(relaxed = true)
+        val dispatcher = mockk<Dispatcher>(relaxed = true)
```

Comment:
[likewise](https://github.com/pusher/chatkit-android/pull/193/files#r387187408)

Reply:
[answered in the initial question](https://github.com/pusher/chatkit-android/pull/193#discussion_r387586462)